src/components/containers/TraceRequiredPanel.js

```diff
@@ -0,0 +1,76 @@
+import React, {Component} from 'react';
+import PanelEmpty from './PanelEmpty';
+import {LayoutPanel} from '../../index';
+import PropTypes from 'prop-types';
+
+class TraceRequiredPanel extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      hasTraces: false,
+    };
+  }
+  checkTraceExistence() {
+    const {visible} = this.props;
+    const {layout} = this.context;
+    const {hasTraces} = this.state;
+
+    if (visible) {
+      /**
+       * Check if there is any trace data
+       */
+      if (
+        Object.keys(layout.xaxis).length === 0 ||
+        Object.keys(layout.yaxis).length === 0
+      ) {
```

Comment:
hmm, I think it's better to look if there's traces by going into _fullData.
there's a key in each trace, called visible, so I think something like this should be pretty accurate:
fullData.filter(trace => trace.visible).length > 0